client/memory.c

```diff
@@ -73,10 +75,10 @@ int xdag_mem_init(size_t size)
 	size++;
 
 	printf("%s , %s\n",g_tmpfile_path, g_tmpname);
-	sprintf(tmpfilename, "%s/%s", g_tmpfile_path, g_tmpname);
+	sprintf(tmpfilename, "%s%s", g_tmpfile_path, g_tmpname);
```

Comment:
The g_tmpfile_path will have "/" "\\" at end or not. Better to check it.

Reply:
Yes, the issue is that the default case is g_tmpfile_path = "", and then this does not work (sorry). 

I agree that better checking can be done, but the change did add more error logging as well, so if sometihgn is wrong it should be made obvius to pool operator at least.

Reply:
I meant not just check, but make it work with such case. Even missing / at the end, we should not report warning or errors, since the path is correct.  Both "./tmp/" and "./tmp" are equal.